utils/utils.go

```diff
@@ -243,3 +243,14 @@ func MovePauseProcessToScope(pausePidPath string) {
 		}
 	}
 }
+
+// Cause returns the most underlying error for the provided one.
+func Cause(err error) error {
```

Comment:
Can we move this into pkg/errorhandling?

Reply:
Done, I also added a max depth to avoid the risk of endless loops.